src/caches/extrapolation_caches.jl

```diff
@@ -47,3 +47,258 @@ function alg_cache(alg::RichardsonEuler,u,rate_prototype,uEltypeNoUnits,uBottomE
+  extrapolation_weights::Array{Rational{BigInt},2}
+  extrapolation_scalars::Array{Rational{BigInt},1}
+  # Weights and scaling factors for internal extrapolation operators (used for error estimate)
+  extrapolation_weights_2::Array{Rational{BigInt},2}
+  extrapolation_scalars_2::Array{Rational{BigInt},1}
+end
+
+function alg_cache(alg::ExtrapolationMidpointDeuflhard,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
+  @unpack n_min, n_init, n_max = alg
+
+  QType = tTypeNoUnits <: Integer ? typeof(qmin_default(alg)) : tTypeNoUnits # Cf. DiffEqBase.__init in solve.jl
+  Q = fill(zero(QType),n_max - n_min + 1)
+
+  n_curr = n_init
+
+  n_old = n_init
+
+  # Initialize subdividing_sequence:
+  if alg.sequence_symbol == :harmonic
+      subdividing_sequence = [BigInt(n+1) for n = 0:n_max]
+  elseif alg.sequence_symbol == :romberg
+      subdividing_sequence = [BigInt(2)^n for n = 0:n_max]
+  else # sequence_symbol == :bulirsch
+      subdividing_sequence = [n==0 ? BigInt(1) : (isodd(n) ? BigInt(2)^Int64(n/2 + 0.5) : 3BigInt(2^Int64(n/2 - 1))) for n = 0:n_max]
+  end
+
+  # Compute stage numbers
+  stage_number = [2sum(Int64.(subdividing_sequence[1:n+1])) - n for n = n_min:n_max]
+
+  # Compute nodes corresponding to subdividing_sequence
+  nodes = BigInt(1) .// subdividing_sequence .^ 2
+
+  # Compute barycentric weights for internal extrapolation operators
+  extrapolation_weights_2 = zeros(Rational{BigInt}, n_max, n_max)
+  extrapolation_weights_2[1,:] = ones(Rational{BigInt}, 1, n_max)
+  for n = 2:n_max
+      distance = nodes[2:n] .- nodes[n+1]
+      extrapolation_weights_2[1:(n-1), n] = extrapolation_weights_2[1:n-1, n-1] .// distance
+      extrapolation_weights_2[n, n] = 1 // prod(-distance)
+  end
+
+  # Compute barycentric weights for extrapolation operators
+  extrapolation_weights = zeros(Rational{BigInt}, n_max+1, n_max+1)
+  for n = 1:n_max
+      extrapolation_weights[n+1, (n+1) : (n_max+1)] = extrapolation_weights_2[n, n:n_max] // (nodes[n+1] - nodes[1])
+      extrapolation_weights[1, n] = 1 // prod(nodes[1] .- nodes[2:n])
+  end
+  extrapolation_weights[1, n_max+1] = 1 // prod(nodes[1] .- nodes[2:n_max+1])
+
+  # Rescale barycentric weights to obtain weights of 1. Barycentric Formula
+  for m = 1:(n_max+1)
+      extrapolation_weights[1:m, m] = - extrapolation_weights[1:m, m] .// nodes[1:m]
+      if 2 <= m
+          extrapolation_weights_2[1:m-1, m-1] = -extrapolation_weights_2[1:m-1, m-1] .// nodes[2:m]
+      end
+  end
+
+  # Compute scaling factors for internal extrapolation operators
+  extrapolation_scalars_2 = ones(Rational{BigInt}, n_max)
+  extrapolation_scalars_2[1] = -nodes[2]
+  for n = 1:(n_max-1)
+      extrapolation_scalars_2[n+1] = -extrapolation_scalars_2[n] * nodes[n+2]
+  end
+
+  # Compute scaling factors for extrapolation operators
+  extrapolation_scalars = -nodes[1] * [BigInt(1); extrapolation_scalars_2]
+
```

Comment:
Because the amount of pre-computations is based on `n_max`, it actually matters in small enough settings. That is something that may be worth remembering to document, or at least test a little bit. BTW, at when `n_max` does overflow start to occur in the sequences?

Reply:
Ok, I will mention this in the documentation.

As far as I've tested the constructor, I have not been able to produce an overflow (for any sequence) if using `BigInt`. As I replaced `BigInt` by `Int`, I got an overflow for the Romberg sequence and `n_max = 6`